unixish.h

```diff
@@ -136,29 +136,31 @@ int afstat(int fd, struct stat *statb);
 #define Mkdir(path,mode)   mkdir((path),(mode))
 
 #if defined(__amigaos4__)
-#  define PERL_SYS_INIT_BODY(c,v)					\
-        MALLOC_CHECK_TAINT2(*c,*v) PERL_FPU_INIT; PERLIO_INIT; MALLOC_INIT; amigaos4_init_fork_array(); amigaos4_init_environ_sema();
-#  define PERL_SYS_TERM_BODY()                         \
-    HINTS_REFCNT_TERM; KEYWORD_PLUGIN_MUTEX_TERM;      \
-    OP_CHECK_MUTEX_TERM; OP_REFCNT_TERM; PERLIO_TERM;  \
-    MALLOC_TERM; LOCALE_TERM; USER_PROP_MUTEX_TERM;    \
-    ENV_TERM;                                          \
-    amigaos4_dispose_fork_array();
+#  define AMIGAOS_SYS_TERM_  amigaos4_dispose_fork_array()
+#  define AMIGAOS_SYS_INIT_ STMT_START {                        \
+                                amigaos4_init_fork_array();     \
+                                amigaos4_init_environ_sema();   \
+                            } STMT_END
+#else 
+#  define AMIGAOS_SYS_TERM_  NOOP
+#  define AMIGAOS_SYS_INIT_  NOOP
 #endif
 
-#ifndef PERL_SYS_INIT_BODY
-#  define PERL_SYS_INIT_BODY(c,v)					\
-        MALLOC_CHECK_TAINT2(*c,*v) PERL_FPU_INIT; PERLIO_INIT; MALLOC_INIT
-#endif
+#define PERL_SYS_INIT_BODY(c,v)					\
+	MALLOC_CHECK_TAINT2(*c,*v) PERL_FPU_INIT; PERLIO_INIT;  \
+        MALLOC_INIT; AMIGAOS_SYS_INIT_;
 
-#ifndef PERL_SYS_TERM_BODY
```

Comment:
I think we still need this `#ifdef`. There are several operating systems that still define their own `PERL_SYS_INIT_BODY` (plan9, dos, vms, os2).

Maybe this can be refactored into some `OS_SYS_INIT_`, I haven't looked into potential ordering issues.